### DISCUSSION TITLE
TypeScript syntax support *without* transformation? (integration with ESBuild)

### DISCUSSION THREAD

**Discussion by mindplay-dk** [babel/babel#13042] [Q&A]
Is there any way to configure Babel with TypeScript support, but *without* the syntax transform?

That is, I want Babel to parse TypeScript syntax, but just emit it without any transform. *TypeScript in, TypeScript out.*

Why, because I'm trying to integrate Babel with ESBuild using `esbuild-plugin-babel` but running into [this issue](https://github.com/nativew/esbuild-plugin-babel/issues/6).

My intuition was, I want the syntax support only, so I'll add `@babel/plugin-syntax-typescript`, which *sounds* like it's going to be just the syntax - it's listed in the documentation as a plugin, even though there's no real documentation for it.

It looks like the syntax plugin is just an implementation detail of the preset though? Probably not meant to be used on it's own. It's not clear to me what it does - it adds/removes some plugins, it seems, but nothing that gives me TypeScript syntax support, it seems.

So perhaps it's not possible?

I guess my only remaining option is a two-step build process, where ESBuild just bundles for modern browsers - so I can have a fast development cycle. And then Babel has to run in a separate build-step for the release-build, where it transforms for IE11 and minifies.

Anyone here had any luck actually integrating ESBuild and Babel? I'd love to hear from you. 😄

**Comment by nicolo-ribaudo** [comment]
The syntax plugin is actually the correct way to do it! Every plugin is "the public api", so you can safely use it.

It disables the `flow` support in the parser and enables the `typescript` support.

**Reply by mindplay-dk** [reply]
Hm, but it doesn't work.

I have this in my configuration:

```
        plugins: [
          "@babel/plugin-syntax-typescript"
        ],
        presets: [
          ['@babel/preset-env', { targets: "ie 11, not ie_mob 11" }]
        ],
```

As soon as it hits a type-hint, it bails:

```
 > src/lib/language.ts:15:40: error: Expected ")" but found ":"
    15 │ export function selectLanguage(available: string[], preferred: string[]): string {
       ╵                                         ^
```

With or without this plugin, there's no difference. 🤷‍♂️

**Reply by nicolo-ribaudo** [reply]
Can you try running your build command with the `BABEL_SHOW_CONFIG_FOR=./src/lib/language.ts` env var? It should print the resolved config for that file.

**Reply by nicolo-ribaudo** [reply]
Also, are you sure that that error is thrown by Babel? What is the complete stack trace? When I try it in the repl I get a different one:
```
Unexpected token, expected "," (1:40)

> 1 | export function selectLanguage(available: string[], preferred: string[]): string {
    |                                         ^
  2 | }
```

https://babeljs.io/repl#?browsers=chrome%2040&build=&builtIns=usage&spec=false&loose=false&code_lz=KYDwDg9gTgLgBAMwK4DsDGMCWEVwM7AA2wGAMgIYoDmS5VwAFOQG7maHkBGxAXPjFEzUA2gF0ANHDBRgCYFBkATPngFCqYgJQq11OAG8AUAF8gA&debug=false&forceAllTransforms=false&shippedProposals=true&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env%2Cstage-1&prettier=false&targets=&version=7.13.12&externalPlugins=

**Reply by nicolo-ribaudo** [reply]
That seems the error format thrown by ESBuild itself, such as in https://github.com/evanw/esbuild/issues/72 :thinking: (different problem but same error)

**Reply by mindplay-dk** [reply]
Aye, it could be ESBuild. No idea. I wish these tools would prefix their messages with the name of the tool. 🤨

**Reply by nicolo-ribaudo** [reply]
Well when the error is thrown by Babel there is a stack trace, so it should make it more evident that the error is thrown by Babel.
